Lots of fixes

abaddonpuff opened the pull request:
Fixed the album ingestion on Spotify and now it guarantees the right Artist to be ingested no more random albums.
Modified and inserted X APIs and now it inserts upon first click
Adjusted the DB to identify the latest post ID 
When no posts are detected, inserts the first 20
Next steps:
    Work on getting data AFTER the latest post when new posts are detected.
    Start working on the Slackbot


<img width="1507" alt="image" src="https://github.com/abaddonpuff/commandCenter/assets/25719314/dae5eb0d-ad1a-4489-ad2d-dfa7b467501b">

abaddonpuff committed:
Lots of fixes

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -35,12 +38,28 @@ def list_twitter(request):
-    breakpoint
+
     #TODO If no posts get last 20 posts from user
+    if len(posts) == 0:
```
You can use the orm: `posts.count()'

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -35,12 +38,28 @@ def list_twitter(request):
+
     #TODO If no posts get last 20 posts from user
+    if len(posts) == 0:
+        last_tweets = get_all_tweets_from_user(user.twitter_handle, max_results=20)
```
I would put the 20 in a constant

bbelderbos replied:
naming: `get_all_tweets_from_user` implies all, but you only get 20 so I would rename this to `get_tweets_from_user`

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -35,12 +38,28 @@ def list_twitter(request):
+                )
 
     #TODO If posts are populated, search for the latest tweet
+        if not created:
```
where does this get set?

bbelderbos replied:
I mean I see created_tweet but that's in the loop

bbelderbos commented on webCommander/centralHub/spotify/spotifyAPI.py:
```diff
@@ -63,7 +63,8 @@ def get_spotify_artist_by_id(artist_id):
 
     response = requests.get(url, headers=spotify_authenticate())
-    return response.text
+    results = json.loads(response.text)
```
requests already has this: 
`response.json()'

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -4,19 +4,22 @@
         form_handle = SubmitXUser(request.POST)
         if form_handle.is_valid():
             x_api_response = form_handle.cleaned_data['handle']
+            breakpoint()
```
remove :)

bbelderbos commented on webCommander/centralHub/views.py:
```diff
@@ -35,12 +38,28 @@ def list_twitter(request):
+        else:
+            user.twitter_last_post_id = last_tweets['meta']['newest_id']
+            messages.success(request, 'Posts Updated')
+            return render(request, 'tweetFramework/handle_summary.html', {'created_tweet':created_tweet})
```
you need to outdent this, if you hit the if here, there's no return = crash

bbelderbos reviewed (changes requested):
Good stuff, feedback ...

bbelderbos commented:
OK reviewed, assuming the db issue is resolved, still stuck on anything at this point?

abaddonpuff committed:
Performed all fixes from previous PR plus added functionality to populate tweets since last tweet.

bbelderbos committed:
fix/add admin
make last post id required at db level as well
removed some unused imports and variables
fixed user var not found syntax
deduplicated some code in the view
added clarifying comment why extra exception handling in one tw api function and the other not

bbelderbos commented:
@abaddonpuff I don't see new commits after last review, do I need to review anything additionally here?
